src/crypto.rs

```diff
@@ -0,0 +1,13 @@
+pub fn sha1_hmac(key: &[u8], payloads: &[&[u8]]) -> [u8; 20] {
+    use hmac::Hmac;
+    use hmac::Mac;
+    use sha1::Sha1;
+
+    let mut hmac = Hmac::<Sha1>::new_from_slice(key).unwrap();
```

Comment:
this unwrap will never hit?

Reply:
The same `unwrap` was present in the previous `From` impl for `Sha1`. It is an internal API, let me see what we can do about it!

Reply:
It's ok to unwrap internally, but there should be a comment of why it is fine.

Reply:
Interesting, I think this can be hit if the provided STUN password is not 20 bytes long. @algesten Any input on this?

Reply:
No, `Hmac` normalizes the key to 20 bytes: https://github.com/RustCrypto/MACs/blob/1a674e4c162222468b13c8b87265f4eda52b6531/hmac/src/lib.rs#L113.